[Alex Chua] iP

AlexCQY opened the pull request:
Complete week 2 increments.

Commit:
Add Gradle support

AlexCQY committed:
Add Greet, Echo, Exit

AlexCQY committed:
Level 2. Add, List

AlexCQY committed:
Level-3: Mark as Done
-Add Task class
-Add mark as done

AlexCQY committed:
Level 4. ToDos, Events, Deadlines
-Add Todo.java, Event.java, Deadline.java which inherits from Task.java. Task is an abstract class.
-Add Enum Command.java to consolidate commands
-Split Duke.java methods. Use of Modularity to manage complexity.

AlexCQY committed:
Level 5. Handle Errors
-Add exception handling for todo, done and invalid commands

AlexCQY committed:
Level 6. Delete
-Add support for delete

AlexCQY committed:
A-TextUiTesting: Automated Text UI Testing
-fixed white spaces error
-test todo,evet,deadline,list, and done

AlexCQY committed:
Update on text-ui-test

AlexCQY committed:
Add empty class DukeException.

AlexCQY committed:
Add save feature for tasks (Level-7)
-task saved at task.txt in current working directory
-add gitignore tasks.txt as it is an output file

AlexCQY committed:
Add LocalDateTime for deadline task (Level-8)

AlexCQY committed:
Merge branch 'branch-Level-7' into master

AlexCQY committed:
Merge branch 'branch-Level-8' into master

# Conflicts:
#	src/main/java/Duke.java

AlexCQY committed:
abstract Ui components to UserInterface.java

AlexCQY committed:
Abstract details from Duke.java to other java classes

AlexCQY committed:
add package duke

AlexCQY committed:
Add test methods to test ToDo and Parser

AlexCQY committed:
create jar file as v0.1 release

AlexCQY committed:
Add some java docs

AlexCQY committed:
add find feature for users to search task description by a keyword

AlexCQY committed:
Configure IDE and code to follow coding style

AlexCQY committed:
Merge branch 'add-gradle-support' of https://github.com/AlexCQY/ip into master

AlexCQY committed:
Change file path of mainClassName in gradle

AlexCQY committed:
Add GUI to Duke using javaFX (Level-10)

ethanso56 commented on src/main/java/duke/UserInterface.java:
```diff
@@ -0,0 +1,146 @@
+import java.util.List;
+
+public class UserInterface {
+    private static final String INDENT = "    ";
```
I like how every message is in a constant.

ethanso56 commented on src/main/java/duke/UserInterface.java:
```diff
@@ -0,0 +1,146 @@
+    private static final String DONE_TASK_MARKED_MESSAGE = "    Nice! I've marked this task as done:";
+    private static final String DELETE_TASK_MARKED_MESSAGE = "    Noted. I've removed this task:";
+
+    private static final String NO_TASK_MESSAGE = "    Sorry the task does not exists";
```
I like how every message is in a constant.

ethanso56 reviewed (commented):
Nice job completing the individual project at this level. In general, I like that the code was easy to read and very clear in what it was trying to achieve. All of the code follows the coding standards well.

Great job!

AlexCQY committed:
Add checkstyle using gradle

AlexCQY committed:
enhance GUI functionality

AlexCQY committed:
Minor bug fix to show invalid message when command not recognise

YuunoKun commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,54 @@
+        );
+        userInput.clear();
+    }
+}
```
I think not having a newline at the end of the file might not pass the checkstyle.
(If you use the IDEA checkstyle plugin, the first line of this .java file will have red lines on it.)

YuunoKun commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Override
+    public String toString() {
+        return "[" + getTaskType() + "]" + "[" + getStatusIcon() + "]" + description + " " + "(by: " + formatDate() + ")";
```
Perhaps you might want to break this line for better readability(Too many characters):
        `return "[" + getTaskType() + "]" + "[" + getStatusIcon() + "]" `
              `+ description + " " + "(by: " + formatDate() + ")";`

YuunoKun commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,255 @@
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+public class Duke  {
```
Missing public class header comments.

YuunoKun commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+            String type = str.substring(1, 2);
+            String status = str.substring(4, 5);
+            String descriptionAndDate = str.substring(6);
+            String[] arr = descriptionAndDate.split("\\(");
```
Perhaps a name that has a better meaning is more suitable for arrays :)

(It has to be plural too)

YuunoKun commented on src/main/java/duke/UserInterface.java:
```diff
@@ -0,0 +1,146 @@
+
+import java.util.List;
+
+public class UserInterface {
```
No header comments for public class and methods in this file. :(

YuunoKun reviewed (commented):
Hi Alex, your code looks good! Just some missing header comments and that's all.   :D

AlexCQY committed:
bug fix - Cannot find tasks.txt file if application opens for the 1st time
Update storage to create file if it does not exists when starting app

AlexCQY committed:
Generate JAR with javaFX packaged in it
Update Mainclassname in gradle to Laucher as it is the main entry point of app using javafx

AlexCQY committed:
code clean up

AlexCQY committed:
A-Assertions
Add some assertions

AlexCQY committed:
Bug fix - accidental delete in previous commit

ethanso56 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,244 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
```
This public class is missing a header comment for javadocs

ethanso56 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+        try {
+            String type = str.substring(1, 2);
+            String status = str.substring(4, 5);
+            String descriptionAndDate = str.substring(6);
```
Might be better to give a plural name for arrays

AlexCQY committed:
clean code by following coding style of cs2103T

AlexCQY committed:
Merge branch 'master' into branch-A-Assertions

AlexCQY committed:
Merge pull request #1 from AlexCQY/branch-A-Assertions

A-Assertions

AlexCQY committed:
Merge branch 'master' into branch-A-CodeQuality

AlexCQY committed:
Merge pull request #2 from AlexCQY/branch-A-CodeQuality

clean code by following coding style of cs2103T

AlexCQY committed:
bugfix - type when resolving conflicts

AlexCQY committed:
Create gradle.yml

AlexCQY committed:
Clean code

AlexCQY committed:
Merge branch 'master' of https://github.com/AlexCQY/ip into master

AlexCQY committed:
Add new command schedule & new class scheduler

AlexCQY committed:
Update event date type from String to LocalDateTime to faciliate scheduling

AlexCQY committed:
-Add Optionals for LocalDateTime as some subclass of task may not have date.
-Add method in scheduler to output tasks that are before a particular date

AlexCQY committed:
Add Category B extension (Viewing Schedule). Current implementation allows user to list tasks before a certain date. Tasks with no date will not be shown.

AlexCQY committed:
Merge pull request #3 from AlexCQY/B-View-Schedule

B view schedule

AlexCQY committed:
Remove IO redirection test

-IO redirection irrelevant to project

AlexCQY committed:
Clean code & add duke greetings

AlexCQY committed:
User guide

AlexCQY committed:
Edit User Guide

AlexCQY committed:
edit user guide

AlexCQY committed:
fix checkstyle issue

AlexCQY committed:
fix checkstyle for CI

AlexCQY committed:
change ui

AlexCQY committed:
add UI.png
